app/deployments/routes.py

```diff
@@ -795,7 +832,11 @@ def updatedep():
         for (k, v) in form_data.items()
         if not k.startswith("extra_opts.")
         and k != "_depid"
-        and (k in stinputs and "updatable" in stinputs[k] and stinputs[k]["updatable"] is True)
```

Comment:
Extremely hard to ready, no clue about what is going on.
`stinputs[k]["updatable"] is True` this is useless, we are checking a boolean expression in a if statement. We can simply write `if (... and stinputs[k]["updatable"] and ...)`

Then, the are really really really to many `and`, but REALLY to many!!!
We can easily loose the control of the branching logic because there are REALLY to many branches.

To simplify, consider to use early returns or something similar, for example:

```python
for k, v in form_data.items() # no braces are required for the tuple
    if "updatable" not in stinputs[k]:
       continue   # since is not updatable we can continue directly to the next item
    # do stuff
```
Moreover, we are evaluating the same conditions (is updatable) twice for nothing